core/src/client/async_client/mod.rs

```diff
@@ -428,42 +461,40 @@ async fn handle_backend_messages<S: TransportSenderT, R: TransportReceiverT>(
 		}
 		// Subscription response.
 		else if let Ok(response) = serde_json::from_slice::<SubscriptionResponse<_>>(&raw) {
-			tracing::debug!("[backend]: recv subscription {:?}", response);
 			if let Err(Some(unsub)) = process_subscription_response(manager, response) {
 				let _ = stop_subscription(sender, manager, unsub).await;
 			}
 		}
 		// Subscription error response.
 		else if let Ok(response) = serde_json::from_slice::<SubscriptionError<_>>(&raw) {
-			tracing::debug!("[backend]: recv subscription closed {:?}", response);
 			let _ = process_subscription_close_response(manager, response);
 		}
 		// Incoming Notification
 		else if let Ok(notif) = serde_json::from_slice::<Notification<_>>(&raw) {
-			tracing::debug!("[backend]: recv notification {:?}", notif);
 			let _ = process_notification(manager, notif);
 		}
 		// Batch response.
 		else if let Ok(batch) = serde_json::from_slice::<Vec<Response<_>>>(&raw) {
-			tracing::debug!("[backend]: recv batch {:?}", batch);
 			if let Err(e) = process_batch_response(manager, batch) {
 				return Err(e);
 			}
 		}
 		// Error response
 		else if let Ok(err) = serde_json::from_slice::<ErrorResponse>(&raw) {
-			tracing::debug!("[backend]: recv error response {:?}", err);
 			if let Err(e) = process_error_response(manager, err) {
 				return Err(e);
 			}
 		}
 		// Unparsable response
 		else {
-			tracing::debug!(
-					"[backend]: recv unparseable message: {:?}",
-					serde_json::from_slice::<serde_json::Value>(&raw)
-				);
-			return Err(Error::Custom("Unparsable response".into()));
+			let json = serde_json::from_slice::<serde_json::Value>(&raw);
```

Comment:
the connection is killed here, we can propagate the entire response in the error then.